tests/test_esel.robot

```diff
@@ -0,0 +1,62 @@
+*** Settings ***
+
+Documentation   This suite is for testing esel's mechanism of checking Reservation_ID.
+
+Resource        ../lib/rest_client.robot
+Resource        ../lib/ipmi_client.robot
+
+Suite Setup            Open Connection And Log In
+Suite Teardown         Close All Connections
+
+*** Test Cases ***
+
+Test Wrong Reservation_ID
+   [Documentation]   This testcase is to test BMC can handle multi-requestor's
+   ...               oem partial add command with incorrect reservation id.
+   ...               It simulates sending partial add command with fake content
+   ...                and wrong Reservation ID. This command will be rejected.
+   
+    ${rev_id_1} =    Run IPMI Command Returned   0x0a 0x42
+    ${rev_id_ls} =   Get Substring   ${rev_id_1}   1   3
+    ${rev_id_ms} =   Get Substring   ${rev_id_1}   -2
+    Run IPMI command   0x0a 0x42
+    ${output} =      Check IPMI Oempartialadd Reject   0x32 0xf0 0x${rev_id_ls} 0x${rev_id_ms} 0 0 0 0 0 1 2 3 4 5 6 7 8 9 0xa 0xb 0xc 0xd 0xe 0xf
+    Should Contain   ${output}   Reservation cancelled
+
+Test Correct Reservation_ID
+   [Documentation]   This testcase is to test BMC can handle multi-requestor's
+   ...               oem partial add command with correct reservation id. It
+   ...                simulates sending partial add command with fake content
+   ...               and correct Reservation ID. This command will be accepted.
+   
+    Run IPMI command   0x0a 0x42
+    ${rev_id_2} =    Run IPMI Command Returned   0x0a 0x42
+    ${rev_id_ls} =   Get Substring   ${rev_id_2}   1   3
+    ${rev_id_ms} =   Get Substring   ${rev_id_2}   -2
+    ${output} =      Check IPMI Oempartialadd Accept   0x32 0xf0 0x${rev_id_ls} 0x${rev_id_ms} 0 0 0 0 0 1 2 3 4 5 6 7 8 9 0xa 0xb 0xc 0xd 0xe 0xf
+    Should Be Empty    ${output}
+
+Clear Test File
+   [Documentation]   Clear /tmp/esel
+   
+   Execute Command   rm /tmp/esel
+   Sleep             10s
+
+*** Keywords ***
+
+Run IPMI Command Returned
+    [arguments]    ${args}
+    ${output_1} =    Execute Command    /tmp/ipmitool -I dbus raw ${args}
+    [return]    ${output_1}
+
+Check IPMI Oempartialadd Reject
+    [arguments]    ${args}
+    ${stdout}    ${stderr}    ${output_2}=  Execute Command    /tmp/ipmitool -I dbus raw ${args}    return_stdout=True    return_stderr= True    return_rc=True
+    [return]    ${stderr}
+
+Check IPMI Oempartialadd Accept
```

Comment:
Hi Rahul,

Please indicate how to change the code to make it merged? 
Thanks.^_^

Regards,

William Li ( Li Nan, ?? ) 
 

Firmware Engineering Professional
OpenPower AE Team | IBM System & Technology Lab
Mobile: +86-186-1081 6605

Beijing, China



From:   Rahul Maheshwari <notifications@github.com>
To:     mkumatag/openbmc-automation 
<openbmc-automation@noreply.github.com>
Cc:     Nan KX Li/China/IBM@IBMCN, Author <author@noreply.github.com>
Date:   05/10/2016 20:26
Subject:        Re: [mkumatag/openbmc-automation] Autotest on Handling 
multiple attempts at ipmi reservation ids -V2 (#50)



In tests/test_esel.robot:
> +   Execute Command   rm /tmp/esel
> +   Sleep             10s
> +
> +*** Keywords ***
> +
> +Run IPMI Command Returned
> +    [arguments]    ${args}
> +    ${output_1} =    Execute Command    /tmp/ipmitool -I dbus raw 
${args}
> +    [return]    ${output_1}
> +
> +Check IPMI Oempartialadd Reject
> +    [arguments]    ${args}
> +    ${stdout}    ${stderr}    ${output_2}=  Execute Command 
/tmp/ipmitool -I dbus raw ${args}    return_stdout=True    return_stderr= 
True    return_rc=True
> +    [return]    ${stderr}
> +
> +Check IPMI Oempartialadd Accept

This keyword seem to be same as above Check IPMI Oempartialadd Reject. We 
can use one keyword here.
?
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub